trino/sqlalchemy/dialect.py

```diff
@@ -238,12 +238,14 @@ def get_indexes(self, connection: Connection, table_name: str, schema: str = Non
             raise exc.NoSuchTableError(f"schema={schema}, table={table_name}")
 
         partitioned_columns = self._get_columns(connection, f"{table_name}$partitions", schema, **kw)
```

Comment:
`$partitions` is specific to some connectors and it's not ensured in the engine level. 

Can you explain how `get_indexes()` is used from caller side? What happens if we don't support this method?

Reply:
Sure. `get_indexes()` is used to get indexes info. There are different kinds of indexes:
* Normal index
![image](https://user-images.githubusercontent.com/6848311/184369175-f0e3f4ab-2a28-460b-be10-f8b24babb5af.png)
* FOREIGN KEY
![image](https://user-images.githubusercontent.com/6848311/184369250-25028a08-3371-4b26-baa4-533849d4544b.png)
* PARTITION in the case of Trino / Presto or Hive

This method could be used by clients like Superset to:
* display info to user (like pictures above)
* perform template query e.g: select data from latest partition.

From Trino driver side, `get_indexes` will call `_get_columns` of `$partitions`, which in turn perform query from `"information_schema"."columns"` table. So, if table is un-partitioned or catalog does not support partition, `_get_columns` just return empty list - no error/exception here.

Reply:
Thanks for sharing the information!

>  So, if table is un-partitioned or catalog does not support partition, _get_columns just return empty list - no error/exception here.

What happens if the 3rd party connectors use `$partitions` table for different purpose and it returns different table definition? Is the current implementation really safe in such situation? 

I know this comment is unrelated to this change, so I'm going to merge this PR. However, please keep in mind that we shouldn't add such connector specific feature to this client.